File: src/main/java/cn/hirrao/javaee/service/BloodPressureService.java
package cn.hirrao.javaee.service;

import cn.hirrao.javaee.entity.BloodPressure;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface BloodPressureService {
    List<BloodPressure> getBloodPressureList(Long uid, String date);

    void insertBloodPressure(long bplId, long userId,  float sbp, float dbp, String recordTime, String classification, String riskLevel);
}

File: api/src/main/kotlin/com/hirrao/javaee/service/BloodPressureService.kt
package com.hirrao.javaee.service

import com.hirrao.javaee.entity.BloodPressure
import com.hirrao.javaee.entity.BloodPressureTable
import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
interface BloodPressureService {
    fun getBloodPressureList(uid: Int, date: LocalDate?): List<BloodPressure?>?

    fun insertBloodPressure(
        bplId: Long,
        userId: Int,
        sbp: Float,
        dbp: Float,
        recordTime: String?,
        classification: String?,
        riskLevel: String?
    )

    fun getBloodPressureTable(date: LocalDate?, uid: Int): BloodPressureTable?

    fun getRiskLevel(uid: Int, date: LocalDate?): List<String?>?
}
